src/lava/lib/optimization/solvers/generic/sub_process_models.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+try:
+    from lava.magma.core.model.nc.net import NetL2
+except ImportError:
+    class NetL2:
+        pass
+
+import numpy as np
+from lava.lib.optimization.solvers.generic.hierarchical_processes import (
+    CostConvergenceChecker,
+    DiscreteVariablesProcess,
+    StochasticIntegrateAndFire)
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.sub.model import AbstractSubProcessModel
+from magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.proc.cost_integrator.process import CostIntegrator
+from lava.proc.dense.process import Dense
+from lava.magma.core.resources import Loihi2NeuroCore
+import numpy as np
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.proc.lif.process import LIF, TernaryLIF
+from lava.proc.scif.process import QuboScif
+
+@implements(proc=DiscreteVariablesProcess, protocol=LoihiProtocol)
+@requires(CPU)
+class DiscreteVariablesModel(AbstractSubProcessModel):

```

Comment:
This seems to be duplicate code of src/lava/lib/optimization/solvers/generic/models.py